Source/include/Ably/ARTRealtimeChannel.h

```diff
@@ -145,6 +151,18 @@ ART_EMBED_INTERFACE_EVENT_EMITTER(ARTChannelEvent, ARTChannelStateChange *)
 
 @end
 
+/**
+ * Describes the properties of the channel state.
+ */
+@interface ARTChannelProperties : NSObject
+/**
+ * Starts unset when a channel is instantiated, then updated with the `channelSerial` from each `ARTChannelEventAttached` event that matches the channel. Used as the value for `ARTRealtimeHistoryQuery.untilAttach`.
+ */
+@property (nonatomic, readonly, nullable) NSString *attachSerial; // CP2a
+@property (nonatomic, readonly, nullable) NSString *channelSerial; // CP2b
```

Comment:
Please could you open a PR in https://github.com/ably/sdk-api-reference/tree/main with a docstring for `channelSerial`, and add it here?

Reply:
`ProtocolMessage` and `ProtocolMessageAction` marked as `internal` in specs:

`class ProtocolMessage: // internal`

Doesn't exist in docstrings on purpose?

Reply:
What does `ProtocolMessage*` have to do with this? This is about `ChannelProperties`, which is a public API.

Reply:
Sorry, forget to expand this closure - the docstring mentioned would be taken from specs and there is that:
```
channelSerial is updated whenever a ProtocolMessage with either MESSAGE, PRESENCE, or ATTACHED actions is received on a channel, and is set to the TR4c channelSerial of that ProtocolMessage, if and only if that field (ProtocolMessage.channelSerial) is populated.
```

Reply:
I mean, it's not a great docstring (since it mentions protocol messages, which as you said are an internal concept), but it's better than nothing.

Reply:
So should I add internal types in docstrings or add it as is? Any explanation of this property without mentioning those internal types will make a little sense.

Reply:
I mean, I don't really know. What would you prefer to do?

Reply:
This is a good question. Looks like the one who edited docstrings has chosen not to put any explanation to `channelSerial`, because `attachSerial` already mentions it. Maybe put there "See `attachSerial` for info"?

Reply:
`attachSerial` doesn't mention it; it mentions the `channelSerial` of "(the) `ARTChannelEventAttached` event" (whatever that means), not that of `ChannelProperties`.

Reply:
https://github.com/ably/ably-cocoa/pull/1965/commits/9d213e4eb4a63202ad4b3c231418c297ef3d5214 and https://github.com/ably/sdk-api-reference/pull/41